src/vcf/record.cpp

```diff
@@ -518,97 +537,70 @@ namespace ebi
-                        message = tag_field + " " + field + " value is not a natural number, negatives not allowed"; 
-                        throw new Error;
-                    }
-                } else if (type == "Float") {
-                    // ...try to cast to float
-                    try {
-                        std::stof(value);
-                    } catch (std::out_of_range) {
-                        // It maybe a subnormal number
-                        std::stold(value);
-                    }
-                } else if (type == "Flag") {
-                    if (value.size() > 1) {
-                        if (status) {
-                            raise(std::make_shared<Error>(line, "There can be only 0 or 1 value"));
-                        } else {
-                            message = tag_field + " " + field + " value is not Flag, it can only have 0 or 1 value";
-                            throw new Error;
-                        }
-                    } else if (value.size() == 1) {
-                        int numeric_value = std::stoi(value);
-                        if (numeric_value != 0 && numeric_value != 1) {
-                            if (status) {
-                                raise(std::make_shared<Error>(line, "A flag must be 0 or 1"));
-                            } else {
-                                message = tag_field + " " + field + " value is not Flag, it must be (1/0/no value)";
-                                throw new Error;
-                            }
-                        }
-                    }
-                    // If no flag is provided then there is nothing to check
-                } else if (type == "Character") {
-                    // ...check the length is 1
-                    if (value.size() > 1) {
-                        raise(std::make_shared<Error>(line, "There can be only one character"));
-                    }
-                } else if (type == "String") {
-                    // ...do nothing, it is guaranteed it will be a string
-                }
-            } catch (...) {
-                if (status) {
-                    raise(std::make_shared<Error>(line, field + " does not match the meta specification Type=" + type));
-                } else {
-                    if (message.size() == 0) {
-                        message = tag_field + " " + field + " value is not " + type;
-                    }
-                    if (tag_field == "INFO") {
-                        throw new InfoBodyError{line, message, field};
-                    } else {
-                        throw new FormatBodyError{line, message};
-                    }
-                }
+                check_value_type(type, value, message);
+            } catch (std::exception &typeError) {
+                raise(std::make_shared<Error>(line, field + " does not match the meta specification Type=" + type + ". " + message));
```

Comment:
I haven't changed the `std::shared_ptr<Error>` to `const std::invalid_argument &` yet! Will be doing it in another commit/PR when this part gets perfected.